### DISCUSSION TITLE
Shipping external OSQuery logs

### DISCUSSION THREAD

**Discussion by hacker0ni** [Security-Onion-Solutions/securityonion#5582] [Unsupported Versions] [answered]
It looks like there are built-in filebeat modules and a 'Security Onion - Osquery' index in ElasticSearch. How can I ship my existing Fleet's result logs to the Security Onion so that I can see and query them in Hunt? So far I've tried pointing a filebeat to the Manager's Logstash and adding a custom Logstash output plugin to a pipeline I wrote, but even when the fields are parsed, I can't search for logs in the Hunt interface and the basic fields like source.ip do not get enriched. Is there an official/recommended way of doing this? The documentation did not help a lot with this, but feel free to point me in the right direction in the docs or your method of achieving this. Thanks a lot everyone!

**Comment by n0th1nt0lOOz** [comment]
Me and some people also asked for adding external log sources but for external integrating is only available for support subscription so almost SO team will not answer kind of those questions clearly.

But if you use distributed architecture, I think you should point filebeat to a sensor node instead

**Reply by defensivedepth** [reply]
@tientmse62290 `for external integrating is only available for support subscription so almost SO team will not answer kind of those questions clearly.`

This is not accurate. Please review the community support guidelines: https://github.com/Security-Onion-Solutions/securityonion/discussions/1720

`Community Support is considered best effort and there are no guarantees and no SLAs. If you need private or priority support, please consider purchasing support from Security Onion Solutions.`

**Reply by n0th1nt0lOOz** [reply]
@defensivedepth very appreciate your working but I just see around 2 or 3 people here to answer for questions. Me and some people asked about the external log source solution a lot but almost receive the document link which is not clear for us to implement. I want to help others but my lack of exp and the document so I can't contribute to community much

**Comment by defensivedepth** [comment, marked as answer]
@Burak-PLT Shipping existing osquery logs into SO can certainly be done, but they will not be parsed correctly by default.

The rough idea would be to have Filebeat pick them up on your Fleet server and send them to the Manager, using this is a Filebeat config section as a guide: https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/filebeat/etc/filebeat.yml#L249 

It would be important to set the `module` correct, as that is what is used in the next step:

Logstash would pick them up on the Manager and send them to the osquery Ingest pipeline ( https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/elasticsearch/files/ingest/osquery.query_result) for final parsing & ingest into SO.

**Reply by hacker0ni** [reply]
Thank you for your answer @defensivedepth ! This really works using the right Filebeat configuration. If we wanted to parse the source.ip fields from the log as well, would we edit the osquery.query_result pipeline? I think the so-osquery index doesn't support all the fields of an index like so-zeek supports. Am I understanding this correctly?

I've found out that the default Logstash pipelines forward the log based on the dataset name (at least for OSQuery). If I generate a clone of osquery.query_result and give it a custom name, what processors would I need to add to that pipeline to achieve source.ip and destination.ip fields in the Hunt?

**Reply by defensivedepth** [reply]
Where are the source.ip fields coming from? What is the original field name? We might be able to include that in the pipeline by default.

**Reply by hacker0ni** [reply]
These fields come from built-in tables like `process_open_sockets` where the source.ip should be mapped to the `local_address` field. We've renamed some of these fields in our queries but even when the original field names are used from the results, these fields are not mapped by default in the Ingest pipelines so we had to go the custom pipeline route. Thank you!

**Reply by defensivedepth** [reply]
Can you create an issue in this repo and reference this Discussion? We can add those tables to the default pipeline

**Reply by hacker0ni** [reply]
Awesome! Please see #5817
